diff --git a/node_modules/@react-native/gradle-plugin/react-native-gradle-plugin/src/main/kotlin/com/facebook/react/tasks/GenerateAutolinkingNewArchitecturesFileTask.kt b/node_modules/@react-native/gradle-plugin/react-native-gradle-plugin/src/main/kotlin/com/facebook/react/tasks/GenerateAutolinkingNewArchitecturesFileTask.kt
index 4c36c8a..8cc91a7 100644
--- a/node_modules/@react-native/gradle-plugin/react-native-gradle-plugin/src/main/kotlin/com/facebook/react/tasks/GenerateAutolinkingNewArchitecturesFileTask.kt
+++ b/node_modules/@react-native/gradle-plugin/react-native-gradle-plugin/src/main/kotlin/com/facebook/react/tasks/GenerateAutolinkingNewArchitecturesFileTask.kt
@@ -61,31 +61,26 @@ abstract class GenerateAutolinkingNewArchitecturesFileTask : DefaultTask() {
             // If user provided a custom cmakeListsPath, let's honor it.
             val nativeFolderPath = sanitizeCmakeListsPath(cmakeListsPath)
             addDirectoryString +=
-                "add_subdirectory(\"$nativeFolderPath\" ${libraryName}_autolinked_build)"
+                "if(EXISTS \"$nativeFolderPath\")\n" +
+                "  add_subdirectory(\"$nativeFolderPath\" ${libraryName}_autolinked_build)\n" +
+                "  list(APPEND AUTOLINKED_LIBRARIES $CODEGEN_LIB_PREFIX$libraryName)\n" +
+                "endif()"
           }
           if (cxxModuleCMakeListsPath != null) {
             // If user provided a custom cxxModuleCMakeListsPath, let's honor it.
             val nativeFolderPath = sanitizeCmakeListsPath(cxxModuleCMakeListsPath)
             addDirectoryString +=
-                "\nadd_subdirectory(\"$nativeFolderPath\" ${libraryName}_cxxmodule_autolinked_build)"
+                "\nif(EXISTS \"$nativeFolderPath\")\n" +
+                "  add_subdirectory(\"$nativeFolderPath\" ${libraryName}_cxxmodule_autolinked_build)\n" +
+                "  if (NOT \"${dep.cxxModuleCMakeListsModuleName}\" STREQUAL \"\")\n" +
+                "    list(APPEND AUTOLINKED_LIBRARIES ${dep.cxxModuleCMakeListsModuleName})\n" +
+                "  endif()\n" +
+                "endif()"
           }
           addDirectoryString
         }
 
-    val libraryModules =
-        packages.joinToString("\n  ") { dep ->
-          var autolinkedLibraries = ""
-          if (dep.libraryName != null) {
-            autolinkedLibraries += "$CODEGEN_LIB_PREFIX${dep.libraryName}"
-          }
-          if (dep.cxxModuleCMakeListsModuleName != null) {
-            autolinkedLibraries += "\n${dep.cxxModuleCMakeListsModuleName}"
-          }
-          autolinkedLibraries
-        }
-
     return CMAKE_TEMPLATE.replace("{{ libraryIncludes }}", libraryIncludes)
-        .replace("{{ libraryModules }}", libraryModules)
   }
 
   internal fun generateCppFileContent(
@@ -160,7 +155,7 @@ abstract class GenerateAutolinkingNewArchitecturesFileTask : DefaultTask() {
     const val COMPONENT_INCLUDE_PATH = "react/renderer/components"
 
     internal fun sanitizeCmakeListsPath(cmakeListsPath: String): String =
-        cmakeListsPath.replace("CMakeLists.txt", "").replace(" ", "\\ ")
+        cmakeListsPath.replace("CMakeLists.txt", "")
 
     // language=cmake
     val CMAKE_TEMPLATE =
@@ -173,11 +168,9 @@ abstract class GenerateAutolinkingNewArchitecturesFileTask : DefaultTask() {
         # or link against a old prefab target (this is needed for React Native 0.76 on).
         set(REACTNATIVE_MERGED_SO true)
 
-        {{ libraryIncludes }}
+        set(AUTOLINKED_LIBRARIES)
 
-        set(AUTOLINKED_LIBRARIES
-          {{ libraryModules }}
-        )
+        {{ libraryIncludes }}
         """
             .trimIndent()
 
